cmd/canary/main.go

```diff
@@ -32,19 +32,30 @@ func main() {
 		err = fmt.Errorf("SAMPLE_INTERVAL is not a valid integer")
 	}
 
+	timeout := 0
+	defaultTimeout := os.Getenv("DEFAULT_MAX_TIMEOUT")
+	if defaultTimeout == "" {
+		timeout = 10
+	} else {
+		timeout, err = strconv.Atoi(defaultTimeout)
+		if err != nil {
+			err = fmt.Errorf("DEFAULT_MAX_TIMOEUT is not a valid integer")
```

Comment:
Reading is hard for me today, and i thought this was canaryd. err is unused in here. I will add a condition on err to have canary stop if set. which should make this right.